stats/memstats/stats.go

```diff
@@ -211,6 +225,69 @@ func (ms *Store) Get(name string, tags stats.Tags) *Measurement {
 	return ms.byKey[ms.getKey(name, tags)]
 }
 
+// GetAll returns the metric for all name/tags register in the store.
+func (ms *Store) GetAll() []Metric {
+	return ms.getAllByName("")
+}
+
+// GetByName returns the metric for each tag variation with the given name.
+func (ms *Store) GetByName(name string) []Metric {
+	if name == "" {
+		panic("name cannot be empty")
+	}
+	return ms.getAllByName(name)
+}
+
+func (ms *Store) getAllByName(name string) []Metric {
+	ms.mu.Lock()
+	defer ms.mu.Unlock()
+
+	metrics := make([]Metric, 0, len(ms.byKey))
+	keys := make([]string, 0, len(ms.byKey))
+	for k, m := range ms.byKey {
+		if name != "" && m.name != name {
+			continue
+		}
+		keys = append(keys, k)
+	}
+	sort.SliceStable(keys, func(i, j int) bool {
+		return keys[i] < keys[j]
+	})
+	for _, key := range keys {
+		m := ms.byKey[key]
+		switch m.mType {
+		case stats.CountType:
+			metrics = append(metrics, Metric{
+				Name:  m.name,
+				Tags:  m.tags,
+				Value: m.LastValue(),
+			})
+		case stats.GaugeType:
+			metrics = append(metrics, Metric{
+				Name:  m.name,
+				Tags:  m.tags,
+				Value: m.LastValue(),
+			})
+		case stats.HistogramType:
+			metrics = append(metrics, Metric{
+				Name:   m.name,
+				Tags:   m.tags,
+				Values: m.Values(),
+			})
+		case stats.TimerType:
+			metrics = append(metrics, Metric{
+				Name:      m.name,
+				Tags:      m.tags,
+				Durations: m.Durations(),
+			})
+		default:
+			panic("unknown measurement type:" + m.mType)
+		}
+	}
+
+	return metrics
+}
```

Comment:
[optional] less code:
```suggestion
func (ms *Store) getAllByName(name string) []Metric {
	ms.mu.Lock()
	defer ms.mu.Unlock()

	keys := lo.Filter(lo.Keys(ms.byKey), func(k string, index int) bool {
		return name == "" || ms.byKey[k].name == name
	})
	sort.SliceStable(keys, func(i, j int) bool {
		return keys[i] < keys[j]
	})
	return lo.Map(keys, func(key string, index int) Metric {
		m := ms.byKey[key]
		switch m.mType {
		case stats.CountType, stats.GaugeType:
			return Metric{
				Name:  m.name,
				Tags:  m.tags,
				Value: m.LastValue(),
			}
		case stats.HistogramType:
			return Metric{
				Name:   m.name,
				Tags:   m.tags,
				Values: m.Values(),
			}
		case stats.TimerType:
			return Metric{
				Name:      m.name,
				Tags:      m.tags,
				Durations: m.Durations(),
			}
		default:
			panic("unknown measurement type:" + m.mType)
		}
	})
}
```